jlm/hls/backend/rvsdg2rhls/hls-function-util.hpp

```diff
@@ -48,6 +48,35 @@ trace_call_rhls(const rvsdg::Output * output);
 std::string
 get_function_name(jlm::rvsdg::Input * input);
 
+// this might already exist somewhere
+template<typename OpType>
+inline const OpType *
+TryGetOwnerOp(const rvsdg::Input & input) noexcept
+{
+  if (const auto node = rvsdg::TryGetOwnerNode<rvsdg::SimpleNode>(input))
+  {
+    return dynamic_cast<const OpType *>(&node->GetOperation());
+  }
+  else
+  {
+    return nullptr;
+  }
+}
+
+template<typename OpType>
+inline const OpType *
+TryGetOwnerOp(const rvsdg::Output & output) noexcept
+{
+  if (const auto node = rvsdg::TryGetOwnerNode<rvsdg::SimpleNode>(output))
+  {
+    return dynamic_cast<const OpType *>(&node->GetOperation());
+  }
+  else
+  {
+    return nullptr;
+  }
+}
+
```

Comment:
I removed all of this in PR #915. Now you are dragging this all in again. The other stuff I rewrote in this PR is also back to the old stuff again. What a mess.

Reply:
@phate This seems to be because we use squashing PRs, which seems to cause git to not fully track the changes correctly. All that's been done is a merge.

Completely agree that this is a mess and that we maybe should stop using squash merges.

Reply:
Removed